src/getRuntimeChoice.ts

```diff
@@ -10,67 +10,69 @@ import * as utils from "./utils";
 // tslint:disable-next-line
 var cjson = require("cjson");
 
+const ENGINE_RUNTIMES: any = {
+  6: "nodejs6",
+  8: "nodejs8",
+  10: "nodejs10",
+};
+
+export const ENGINES_FIELD_REQUIRED_MSG = clc.bold(
```

Comment:
Looks like you're exporting these so you can use them in tests. In general if you're doing something just so you can test it's a kind of smell...

I think the `to.throw()` is pretty loose in how it looks at error messages (simple includes or regex or something). You could maybe not export these and just do something like `to.throw(FirebaseError, "unsupported")`?

Reply:
Yeah, these are exported for tests - this makes it so the error message is all in one place, so if the message itself changes, the tests will not break since the message is defined in one place. So, if something changes in the string to modify "unsupported" to, say, "not supported", then the tests would break even though functionality has remained the same.

That being said, I'm not super happy with the structure of this code and tests - I feel like these functions should all be tested separately and should not be throwing or logging warnings, but should be returning something sensible that the test can then check, but so little is happening here that it's not worth creating all these extra wrapper functions just to make it more modularized. So, I'm rather torn.

Reply:
I'd lean toward not exporting even if it makes the test brittle against the language. If we change the language, the test will fail and we'll have to change it but that's very straightforward. Exporting the error messages just for testing ends up feeling like an overkill solution for a quickly fixed problem.

Reply:
Went with keeping the constants https://github.com/firebase/firebase-tools/pull/1223#discussion_r280012336. But if there are strong feelings against, I don't mind changing.